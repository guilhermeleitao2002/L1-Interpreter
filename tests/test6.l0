let x = 10;
let makeCounter = fn => {
  let count = box 0;
  fn => {
    count := !count + 1;
    !count + x
  }
};

let counter1 = ((makeCounter) ());
let counter2 = ((makeCounter) ());

println ((counter1) ());  // 11 (1 + 10)
println ((counter1) ());  // 12 (2 + 10)
println ((counter2) ());  // 11 (1 + 10)
println ((counter1) ());  // 13 (3 + 10)
println ((counter2) ());;  // 12 (2 + 10)