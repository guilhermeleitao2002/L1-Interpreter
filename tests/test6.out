11
12
11
13
12